telegram/ext/_application.py

```diff
@@ -397,6 +405,11 @@ async def initialize(self) -> None:
             return
 
         await self.bot.initialize()
+        if isinstance(self._update_processor, BaseUpdateProcessor) and not isinstance(
+            self._update_processor, SimpleUpdateProcessor
+        ):
+            await self._update_processor.initialize()
```

Comment:
`self._update_processor` always is an instance of `BUP`, isn't it? And we should just always call `initialize`, no matter which exact implementation is used :) If the implementation decides that `initialize` doesn't do anything that's fine, but that's not the concern of the `Application` class.

Same for `shutdown` :)

Reply:
Well, its always an instance of `BUP` but when nothing is passed it uses `SUP` so we need to check it with both `SUP` and `BUP`. The `call_after` only checks upto one stacklevel, i.e. it will be call `SUP.initialize` when `SUP` object is passed and `BUP.initialize` when `BUP` object is passed (same for shutdown ofc), so we have to write seperate tests for both cases thats why I added this implementation. :thinking:

Reply:
@Bibo-Joshi

Reply:
Let me point you to https://github.com/python-telegram-bot/python-telegram-bot/pull/3654#discussion_r1186681941 ;P If the tests don't work as disered, let's adapt the tests, not the code ;) TBH I don't quite understand where the issue in the tests is, though …